### ISSUE TITLE
expand NetherEx config support

### ISSUE THREAD

**Issue by jsanchezi** [jbredwards/Nether-API#7]
So basically the only way to easily create custom nether biomes with json is by creating a biome with biometweaker and then making it generate with netherex which has a ton of config options to customize your biome. But for somo reason when nether api is installed, it doesn't work. It somehow messes  up netherex's generation.

**Comment by jbredwards** [comment]
Nether API overrides NetherEx's biome gen system, so you have to make your custom biomes generate via Nether API instead. You can do this through a combination of [`GroovyScript`](https://modrinth.com/mod/groovyscript) and the [`NetherAPIRegistryEvent.Nether`](https://github.com/jbredwards/Nether-API/blob/823815229c3cb8812ea0f7fa8b51bf8827ee1ec4/src/main/java/git/jbredwards/nether_api/api/event/NetherAPIRegistryEvent.java#L78) event:
```groovy
import git.jbredwards.nether_api.api.event.NetherAPIRegistryEvent

event_manager.listen { NetherAPIRegistryEvent.Nether event ->
    event.registry.registerBiome(biome("YOUR_BIOME_ID"), YOUR_BIOME_WEIGHT)
}
```
You can read more about how GroovyScript events work on their official wiki: https://cleanroommc.com/groovy-script/minecraft/events. FYI, I think GroovyScript also lets you register custom biomes, so you might not need BiomeTweaker (unless you're also using it for other things).

**Comment by jbredwards** [comment]
Actually, I'm looking into this a bit further and it looks like something I can add support for without it breaking anything :)

**Comment by jsanchezi** [comment]
The problem is you need to know java to be able to do that. This is what netherex allowed so it would be amazing if either nether api was more easily customizable or that netherex was somehow allowed to generate new biomes overwriting nether api. Thanks for hearing me out.
![image](https://github.com/user-attachments/assets/b44100d3-79b1-4609-b952-14a1d0bf2acc)

**Comment by jbredwards** [comment]
> The problem is you need to know java to be able to do that.

Yes, but actually no :P. I provided everything you need in the comment above (that you can just copy/paste). Nether API already supports the rest of NetherEx's biome data config settings, like entities, traits, and everything else that isn't biome weight (yet), so you can just leave those settings alone and they'll generate fine.

> This is what netherex allowed so it would be amazing if either nether api was more easily customizable or that netherex was somehow allowed to generate new biomes overwriting nether api.

The way I'm thinking to add this is by replacing the current hardcoded registry [here](https://github.com/jbredwards/Nether-API/blob/823815229c3cb8812ea0f7fa8b51bf8827ee1ec4/src/main/java/git/jbredwards/nether_api/mod/common/compat/netherex/NetherExHandler.java#L37-L47) with one that goes through all biomes, checks if they're registered in NetherEx's biome data handler, then copying the weight data to Nether API's registry for each one that is.

**Comment by jsanchezi** [comment]
and how do you change fog colour and all the other options netherex has with nether api? In netherex it's as simple as above but in nether api is probably quite complicated. Do you have an example?

**Comment by jbredwards** [comment]
Fixed in v1.4.0!
